### PR TITLE
UI bugfixes

### PR DESCRIPTION
**Pull request by zathras-crypto**
This PR contains further bugfixes picked up from regtesting.

### CODE REVIEW

**Comment by zathras-crypto** [comment]
NOTE: Not to be merged yet, there is a hang now occurring when running my regtest scenarios.
```diff
diff --git a/src/qt/metadexdialog.cpp b/src/qt/metadexdialog.cpp
index ce4d997..f970ed1 100755
--- a/src/qt/metadexdialog.cpp
+++ b/src/qt/metadexdialog.cpp
@@ -123,7 +123,7 @@ void MetaDExDialog::setWalletModel(WalletModel *model)
 
 void MetaDExDialog::PopulateAddresses()
 {
-    LOCK(cs_tally);
+//    LOCK(cs_tally);
 
     uint32_t propertyId = global_metadex_market;
     bool testeco = false;
```

Seems to fix it but that is very likely unsafe and there is probably a lock order issue here.  Recompiling with -DDEBUG_LOCKORDER now and will see what I can find.

**Comment by zathras-crypto** [comment]
```
2015-06-26 07:53:21 POTENTIAL DEADLOCK DETECTED
2015-06-26 07:53:21 Previous lock order was:
2015-06-26 07:53:21  (1) cs_tally  qt/metadexdialog.cpp:126
2015-06-26 07:53:21  (2) cs_main  omnicore/omnicore.cpp:160
2015-06-26 07:53:21 Current lock order is:
2015-06-26 07:53:21  (2) cs_main  main.cpp:3123
2015-06-26 07:53:21  (2) cs_main  main.cpp:2213
2015-06-26 07:53:21  (1) cs_tally  omnicore/omnicore.cpp:319
2015-06-26 07:53:21 POTENTIAL DEADLOCK DETECTED
2015-06-26 07:53:21 Previous lock order was:
2015-06-26 07:53:21  (1) cs_tally  qt/metadexdialog.cpp:126
2015-06-26 07:53:21  cs_main  omnicore/omnicore.cpp:2130
2015-06-26 07:53:21  (2) wallet->cs_wallet  omnicore/omnicore.cpp:2130
2015-06-26 07:53:21 Current lock order is:
2015-06-26 07:53:21  cs_main  rpcserver.cpp:1052
2015-06-26 07:53:21  (2) pwalletMain->cs_wallet  rpcserver.cpp:1052
2015-06-26 07:53:21  (1) cs_tally  omnicore/omnicore.cpp:319
```

**Comment by zathras-crypto** [comment]
@dexX7 I have the debug info, mind if I ask what diagnostic process you use for determining where the problem lies?

**Comment by dexX7** [comment]
Ah, it looks like you recreated the issue I tried to solve earlier.

In addition to the deadlock warning, it's possible to log all lock entries and releases, but earlier this wasn't really useful.
```
./configure CXXFLAGS="-DDEBUG_LOCKORDER -DDEBUG_LOCKCONTENTION -g"
```

I'm not entirely sure about this one, and I assume there is another solution, but I noticed that locking `cs_tally` in `PopulateAddresses()` and then calling `UpdateBalances() -> UpdateBuyAddressBalance() -> feeCheck()` doesn't go well.

Fix following inline.

File: src/qt/metadexdialog.cpp [hunk 1]
```diff
@@ -107,21 +107,78 @@ void MetaDExDialog::setClientModel(ClientModel *model)
...
+    LOCK(cs_tally);
+
+    uint32_t propertyId = global_metadex_market;
+    bool testeco = false;
+    if (propertyId >= TEST_ECO_PROPERTY_1) testeco = true;
+
+    // get currently selected addresses
+    QString currentSetBuyAddress = ui->buyAddressCombo->currentText();
+    QString currentSetSellAddress = ui->sellAddressCombo->currentText();
+
```
Comment (dexX7) [inline src/qt/metadexdialog.cpp]: Start here with:
```c++
{
    LOCK(cs_tally);
```

File: src/qt/metadexdialog.cpp [hunk 2]
```diff
@@ -107,21 +107,78 @@ void MetaDExDialog::setClientModel(ClientModel *model)
...
+    // clear address selectors
+    ui->buyAddressCombo->clear();
+    ui->sellAddressCombo->clear();
+
```
Comment (dexX7) [inline src/qt/metadexdialog.cpp]: Start here with:
```c++
{
    LOCK(cs_tally);
```

File: src/qt/metadexdialog.cpp [hunk 3]
```diff
@@ -107,21 +107,78 @@ void MetaDExDialog::setClientModel(ClientModel *model)
...
+                if (!getUserAvailableMPbalance(address, OMNI_PROPERTY_MSC)) continue;
+                if (IsMyAddress(address) == ISMINE_SPENDABLE) ui->buyAddressCombo->addItem((my_it->first).c_str());
+            }
+            if (id == OMNI_PROPERTY_TMSC && testeco) {
+                if (!getUserAvailableMPbalance(address, OMNI_PROPERTY_TMSC)) continue;
+                if (IsMyAddress(address) == ISMINE_SPENDABLE) ui->buyAddressCombo->addItem((my_it->first).c_str());
+            }
+        }
+    }
+
```
Comment (dexX7) [inline src/qt/metadexdialog.cpp]: Close the scope here with:
```c++
}
```

This limits the scope of the lock.

Comment (zathras-crypto) [inline src/qt/metadexdialog.cpp]: Ahhh, I was wondering why you'd enclosed the previous address population code in ```{ }```, I thought that was a no-op but now it makes complete sense.

Thanks dude! :)  Confirmed working.

**Comment by zathras-crypto** [comment]
Oh, that's weird - my comment disappeared :(

> Ahhh, I was wondering why you'd enclosed the previous address population code in { }, I thought that was a no-op but now it makes complete sense.
Thanks dude! :) Confirmed working.

**Comment by dexX7** [comment]
Ah sorry, I probably should have posted a bit more of explanation - these `{ }` act as "fence".

**Comment by zathras-crypto** [comment]
Yeah I figured as much, thanks heaps mate :)

**Comment by zathras-crypto** [comment]
Should be ok to merge now I believe (if you have no objections bud)

**Pull request merged**